src/fileops/httpiochain.cpp

```diff
@@ -131,9 +131,26 @@ dav_ssize_t HttpIOChain::write(IOChainContext & iocontext, const void *buf, dav_
     CHAIN_FORWARD(write(iocontext, buf, count));
 }
 
+void HttpIOChain::writeFromBuffer(IOChainContext&  iocontext, const char* buff,
+                                  dav_size_t size, const std::string& uploadId,
+                                  std::vector<std::string>& etags,
+                                  int partNumber) {
+    CHAIN_FORWARD_VOID(writeFromBuffer(iocontext, buff, size, uploadId,
+                       etags, partNumber));
+}
+
 dav_ssize_t HttpIOChain::writeFromProvider(IOChainContext & iocontext, ContentProvider &provider) {
     CHAIN_FORWARD(writeFromProvider(iocontext, provider));
 }
 
+std::string HttpIOChain::initiateMultipart(IOChainContext& iocontext) {
+     CHAIN_FORWARD(initiateMultipart(iocontext));
+}
+
+void HttpIOChain::commitChunks(IOChainContext& iocontext,
+                               const std::string &uploadId,
+                               const std::vector<std::string> &etags) {
+     CHAIN_FORWARD_VOID(commitChunks(iocontext, uploadId, etags));
+}
```

Comment:
Not really enthusiastic about these, as NOOPs have been added to the top level HTTPIOChain interface :confused: 

Same applies for the `HttpIOChain::writeFromBuffer(..)` function

Reply:
Well, I don't see why you think they are NOOPS. This is the architecture and the only way to get to the right protocol object that implements the stated functon is to chain forward to it. If you remove them multi-part stops working. I will agree it's a poor artchitecture but I don't think we have the resources to reqwite it.